src/main/java/com/zufar/onlinestore/payment/exception/handler/PaymentExceptionsHandler.java

```diff
@@ -0,0 +1,78 @@
+        ApiResponse<String> apiResponse = buildResponse(exception, HttpStatus.NOT_FOUND);
+        log.error("Handle payment not found exception: failed: message: {}, description: {}.", apiResponse.message(), apiResponse.data());
+        return apiResponse;
+
+    }
+
+    @ExceptionHandler(PaymentEventProcessingException.class)
+    @ResponseStatus(HttpStatus.BAD_REQUEST)
+    public ApiResponse<String> handlePaymentEventProcessingException(final PaymentEventProcessingException exception) {
+        ApiResponse<String> apiResponse = buildResponse(exception, HttpStatus.BAD_REQUEST);
+        log.error("Handle payment event processing exception: failed: message: {}, description: {}.", apiResponse.message(), apiResponse.data());
+        return apiResponse;
+    }
+
+    @ExceptionHandler(PaymentIntentProcessingException.class)
+    @ResponseStatus(HttpStatus.BAD_REQUEST)
+    public ApiResponse<String> handlePaymentIntentProcessingException(final PaymentIntentProcessingException exception) {
+        ApiResponse<String> apiResponse = buildResponse(exception, HttpStatus.BAD_REQUEST);
+        log.error("Handle payment intent processing exception: failed: message: {}, description: {}.", apiResponse.message(), apiResponse.data());
+        return apiResponse;
+    }
+
+    @ExceptionHandler(PaymentMethodProcessingException.class)
+    @ResponseStatus(HttpStatus.BAD_REQUEST)
+    public ApiResponse<String> handlePaymentMethodProcessingException(final PaymentMethodProcessingException exception) {
+        ApiResponse<String> apiResponse = buildResponse(exception, HttpStatus.BAD_REQUEST);
+        log.error("Handle payment method processing exception: failed: message: {}, description: {}.", apiResponse.message(), apiResponse.data());
+        return apiResponse;
+    }
+
+
+    @ExceptionHandler(PaymentEventParsingException.class)
+    @ResponseStatus(HttpStatus.BAD_REQUEST)
+    public ApiResponse<String> handlePaymentEventParsingException(final PaymentEventParsingException exception) {
+        ApiResponse<String> apiResponse = buildResponse(exception, HttpStatus.BAD_REQUEST);
+        log.error("Handle payment event parsing exception: failed: message: {}, description: {}.", apiResponse.message(), apiResponse.data());
+        return apiResponse;
+    }
+
+    private ApiResponse<String> buildResponse(Exception exception, HttpStatus httpStatus) {
+        return ApiResponse.<String>builder()
+                .data(getExceptionDescription(exception))
+                .message(exception.getMessage())
+                .timestamp(LocalDateTime.now())
+                .httpStatusCode(httpStatus.value())
+                .build();
+    }
+
+    private String getExceptionDescription(Exception exception) {
+        return Arrays.stream(exception.getStackTrace())
+                .findFirst()
+                .map(topElement -> DESCRIPTION_TEMPLATE.formatted(topElement.getMethodName(), topElement.getClassName()))
+                .orElseGet(String::new);
```

Comment:
To clarify, the orElse method will indeed evaluate its argument regardless of whether the Optional is empty or not. This doesn't usually matter if the argument is a constant or a precomputed value, like "". But it does matter if the argument is a method call or an expression with side effects, especially if those are potentially costly in terms of performance.

so I believe its better to use OrElse(StringUtils.EMPTY) here

Reply:
done